tests/suite/tracing_test.go

```diff
@@ -0,0 +1,263 @@
+		defer cancel()
+
+		key := types.NamespacedName{Name: gatewayClassName}
+		var gwClass gatewayv1.GatewayClass
+		Expect(k8sClient.Get(ctx, key, &gwClass)).To(Succeed())
+
+		gwClass.Spec.ParametersRef = nil
+
+		Expect(k8sClient.Update(ctx, &gwClass)).To(Succeed())
+	})
+
+	updateGatewayClass := func() error {
+		ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.CreateTimeout)
+		defer cancel()
+
+		key := types.NamespacedName{Name: gatewayClassName}
+		var gwClass gatewayv1.GatewayClass
+		if err := k8sClient.Get(ctx, key, &gwClass); err != nil {
+			return err
+		}
+
+		gwClass.Spec.ParametersRef = &gatewayv1.ParametersReference{
+			Group: ngfAPI.GroupName,
+			Kind:  gatewayv1.Kind("NginxProxy"),
+			Name:  "nginx-proxy",
+		}
+
+		return k8sClient.Update(ctx, &gwClass)
+	}
+
+	sendRequests := func(url string, count int) {
+		for range count {
+			Eventually(
+				func() error {
+					status, _, err := framework.Get(url, address, timeoutConfig.RequestTimeout)
+					if err != nil {
+						return err
+					}
+					if status != http.StatusOK {
+						return fmt.Errorf("status not 200; got %d", status)
+					}
+					return nil
+				}).
+				WithTimeout(timeoutConfig.RequestTimeout).
+				WithPolling(500 * time.Millisecond).
+				Should(Succeed())
+		}
+	}
+
+	// Send traffic and verify that traces exist for hello app. We send every time this is called because
+	// sometimes it takes awhile to see the traces show up.
+	findTraces := func() bool {
+		sendRequests(helloURL, 25)
+		sendRequests(worldURL, 25)
+		sendRequests(helloworldURL, 25)
+
+		logs, err := resourceManager.GetPodLogs(framework.CollectorNamespace, collectorPodName, &core.PodLogOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		return strings.Contains(logs, "service.name: Str(ngf:helloworld:gateway:my-test-svc)")
+	}
+
+	checkStatus := func() {
+		Eventually(
+			func() error {
+				return verifyGatewayClassResolvedRefs()
+			}).
+			WithTimeout(timeoutConfig.GetTimeout).
+			WithPolling(500 * time.Millisecond).
+			Should(Succeed())
+
+		Eventually(
+			func() error {
+				return verifyPolicyStatus()
+			}).
+			WithTimeout(timeoutConfig.GetTimeout).
+			WithPolling(500 * time.Millisecond).
+			Should(Succeed())
+
+		// wait for expected first line to show up
+		Eventually(findTraces, "1m", "5s").Should(BeTrue())
+	}
+
+	It("sends tracing spans for one policy attached to one route", func() {
+		sendRequests(helloURL, 5)
+
+		// verify that no traces exist yet
+		logs, err := resourceManager.GetPodLogs(framework.CollectorNamespace, collectorPodName, &core.PodLogOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(logs).ToNot(ContainSubstring("service.name: Str(ngf:helloworld:gateway:my-test-svc)"))
+
+		// install tracing configuration
+		traceFiles := []string{
+			nginxProxyFile,
+			policySingleFile,
+		}
+		Expect(resourceManager.ApplyFromFiles(traceFiles, ns.Name)).To(Succeed())
+		Expect(updateGatewayClass()).To(Succeed())
+
+		checkStatus()
```

Comment:
I suggest renaming to checkStatusAndTraces.
got a bit confused by reading the test about when we actually check for traces